lib/remap-lang/grammar.pest

```diff
@@ -1,7 +1,7 @@
 // Root ------------------------------------------------------------------------
 
 program     = _{ SOI ~ NEWLINE* ~ expressions ~ NEWLINE* ~ EOI }
-expressions = _{ (expression ~ (EOE+ ~ expression)*)+ ~ EOE? }
+expressions = _{ expression ~ (EOE+ ~ expression)* ~ EOE? }
```

Comment:
This is what caused [this failure](https://github.com/timberio/vector/pull/5856#discussion_r551862899) and others. I'm not sure how this sneaked in here, and we definitely had a gap in our test coverage here.

I've fixed this, re-enabled the tests that started failing, and added extra tests to make sure we catch a regression like this in the future.